### ISSUE TITLE
[Linux] GoTo definition performs symlink resolution

### ISSUE THREAD

**Issue by AhmedKamal20** [microsoft/pylance-release#5136]
Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual

- I Have `/mnt/extra-drive/Git` that contains multiple projects
- I symlinked this directory to `~/Git`
  - `ln -s /mnt/extra-drive/Git ~/Git`
- And I open the projects using vscode from this new directory `~/Git`
- When I open any file from the sidebar it opens from `~/Git/project` as normal
- But if I try to "Go to Defenition" from class, it redirects me to the file from `/mnt/extra-drive/Git/project`, I want it to open from `~/Git/project` to reflect on the sidebar "Explorer View" correctly

## Steps to reproduce:

1. Symlink a Python project to another Path
2. Open this new Path on VsCode
3. Try to perform GoTo definition
4. It will open the file from the original path, and it will not reflect on the explorer that you open the file 

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.10.12
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): PipEnv
-   Value of the `python.languageServer` setting: Default

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

```
2023-11-21 17:35:29.624 [info] Experiment 'pythonPromptNewFormatterExt' is active
2023-11-21 17:35:29.624 [info] Experiment 'pythonPromptNewToolsExt' is active
2023-11-21 17:35:29.624 [info] Experiment 'pythonREPLSmartSend' is active
2023-11-21 17:35:29.624 [info] Experiment 'pythonTestAdaptercf' is active
2023-11-21 17:35:29.624 [info] Default formatter is set to null for workspace /home/akamal/Git/XXX/XXX-2-dev/project
2023-11-21 17:35:29.624 [info] Test server listening.
2023-11-21 17:35:29.624 [info] Python interpreter path: ~/.local/share/virtualenvs/XXX-2-dev-5aTy5fI8/bin/python
2023-11-21 17:35:33.946 [info] Starting Pylance language server.
```

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```

languageServer: "Pylance"

```

</p>
</details>

Extension version: 2023.20.0
VS Code version: Code 1.84.2 (1a5daa3a0231a0fbba4f14db7ec463cf99d7768e, 2023-11-09T10:50:47.800Z)
OS version: Linux x64 6.2.0-36-generic
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 5900HX with Radeon Graphics (16 x 4161)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|0, 1, 1|
|Memory (System)|30.69GB (14.00GB free)|
|Process Argv|--unity-launch --crash-reporter-id 7f93fee5-08b2-4e19-979d-5a659b937db7|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu-xorg|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu-xorg|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263cf:30880073
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
2e4cg342:30602488
f6dab269:30613381
showlangstatbar:30737416
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxpt:30866567
pythonnoceb:30805159
dsvsc013:30795093
dsvsc014:30804076
dsvsc015:30845448
pythontestfixt:30871694
pythonregdiag2:30871582
pyreplss2:30886141
pythonmypyd1:30879173
pythoncet0:30885854
pythontbext0:30879054
accentitlementsc:30887149
dsvsc016cf:30886111
dsvsc017:30886112
dsvsc018:30886114
aa_t_chat:30882232

```

</details>

**Comment by debonte** [comment]
Dupe of https://github.com/microsoft/pylance-release/issues/5030?

**Comment by rchiodo** [comment]
I can't seem to reproduce on windows. I'll try on WSL.

**Comment by rchiodo** [comment]
Yeah it reproduces on WSL. So a linux only problem at the moment. Might be a dupe of 5030.

**Comment by starball5** [comment]
Related on Stack Overflow: ['Go to Definition' from a Python file in a symlinked directory redirects me to the canonical path](https://stackoverflow.com/q/77510981/11107541)

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2023.11.102, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202312102-1-december-2023-prerelease)
